cmd/geth/config.go

```diff
@@ -115,12 +129,28 @@ func defaultNodeConfig() node.Config {
 	return cfg
 }
 
+func metricsFromCliArgs(ctx *cli.Context) metricConfig {
+	return metricConfig{
+		Enabled:          ctx.GlobalBool(utils.MetricsEnabledFlag.Name),
+		EnabledExpensive: ctx.GlobalBool(utils.MetricsEnabledExpensiveFlag.Name),
+		HTTP:             ctx.GlobalString(utils.MetricsHTTPFlag.Name),
+		Port:             ctx.GlobalInt(utils.MetricsPortFlag.Name),
+		EnableInfluxDB:   ctx.GlobalBool(utils.MetricsEnableInfluxDBFlag.Name),
+		InfluxDBEndpoint: ctx.GlobalString(utils.MetricsInfluxDBEndpointFlag.Name),
+		InfluxDBDatabase: ctx.GlobalString(utils.MetricsInfluxDBDatabaseFlag.Name),
+		InfluxDBUsername: ctx.GlobalString(utils.MetricsInfluxDBUsernameFlag.Name),
+		InfluxDBPassword: ctx.GlobalString(utils.MetricsInfluxDBPasswordFlag.Name),
+		InfluxDBTags:     ctx.GlobalString(utils.MetricsInfluxDBTagsFlag.Name),
+	}
+}
+
 // makeConfigNode loads geth configuration and creates a blank node instance.
 func makeConfigNode(ctx *cli.Context) (*node.Node, gethConfig) {
 	// Load defaults.
 	cfg := gethConfig{
-		Eth:  eth.DefaultConfig,
-		Node: defaultNodeConfig(),
+		Eth:     eth.DefaultConfig,
+		Node:    defaultNodeConfig(),
+		Metrics: metricsFromCliArgs(ctx),
```

Comment:
I don't understand why you need this function (`metricsFromCliArgs`) if you also have `applyMetricConfig`. The idea behind the config file loading is:

The config is always created using the Go code defaults (on line 150).
Then, the TOML file is loaded to override some things.
Finally, the flags are applied to override more things.

In this PR, for metrics, it takes the defaults from flags, then overrides with config file, then applies the flags again.
I think we should remove metricsFromCliArgs and just define the defaults for metrics directly in the code.

Reply:
Yeah, the problem with this approach is that we have now two different places where the default value of the flags is defined.
Once here and once in the cli.Flags.

Reply:
The way we usually solve this is by taking the flag default from the Go struct default. 

https://github.com/ethereum/go-ethereum/blob/master/cmd/utils/flags.go#L136

Reply:
I created a new default var in the `metrics` package